sherpa/plot/__init__.py

```diff
@@ -2711,9 +2711,9 @@ def contour(self, overplot=False, clearwindow=True, **kwargs):
 
         # Note: the user arguments are not applied to the point
         #
-        point = Point()
-        point.point_prefs = self.point_prefs
-        point.point(self.parval0, self.parval1,
+        point = Plot()
```

Comment:
So `Plot` is clever enough to recognize it's been sent scalar values for x and y and internally converts it into an array. Or does this rely on matplotlib's handlling of scalar values here (in which case we'd want to ensure `Plot` handles scalar values sensibly otherwise it depends on the backend).

Reply:
We will require `plot` in the backend to accept scalars or arrays (but documenting and testing those requirements is a different PR that I have not yet opened), so the onus is on the backend module. Both matplotlib and bokeh accept scalars out of the box with no extra work needed. But if we ever implement a backend module that does not, then the `crazy_backend.plot` code would have to take care of converting it to lists or arrays or whatever the datastructure for `crazy_backend` is.

Reply:
I think I wanted to surface whether this was already, or is going to be, something that could rely on the backend or is something we support (i.e. sending in a scalar value to `Plot`). I think it should be the latter. Given that you are changing this in the PR I think there should be s very simple change to the `Plot.plot` docs along the lines of changing

```
    def plot(self, x, y, yerr=None, xerr=None, title=None, xlabel=None,
             ylabel=None, overplot=False, clearwindow=True,
             **kwargs):
        """Plot the data.

        Parameters
        ----------
        x, y
           The data values to plot. They should have the same length.
        yerr, xerr : optional
           The symmetric errors to apply to the y or x values, or `None`
           for no errors along the axis. If given, they should match the
           length of the data.
```

to mention `x`/`y` can be scalars (and `xerr`/`yerr`).

Reply:
Note: I also will (in a later PR) synchronize doctrings between these objects (for example, many functions in `pylab_backend` have incomplete docstrings right now), but that's no excuse to not do this properly here already. 
Done in last commit.